azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -15,24 +15,48 @@
 # Requires Python 2.4+ and Openssl 1.0+
 #
 
+import json
+import os
 from azurelinuxagent.common import logger
 from azurelinuxagent.common.event import WALAEventOperation, add_event
 from azurelinuxagent.common import conf
 from azurelinuxagent.common.exception import AgentError
+from azurelinuxagent.common.protocol.extensions_goal_state_from_vm_settings import _CaseFoldedDict
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
+
+
+# Customer-defined policy is expected to be located at this path.
+# If there is no file at this path, default policy will be used.
+CUSTOM_POLICY_PATH = DefaultOSUtil().get_custom_policy_file_path()
+
+# Default policy values to be used when no custom policy is present.
+_DEFAULT_ALLOW_LISTED_EXTENSIONS_ONLY = False
+_DEFAULT_SIGNATURE_REQUIRED = False
 
 
 class PolicyError(AgentError):
     """
     Error raised during agent policy enforcement.
     """
+    # TODO: split into two error classes for internal/dev errors and user errors.
+    def __init__(self, msg=None, inner=None, code=-1):
+        super(PolicyError, self).__init__(msg, inner)
+        self.code = code
 
 
-class PolicyEngine(object):
+class _PolicyEngine(object):
     """
     Implements base policy engine API.
     """
-    @classmethod
-    def _log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
+    def __init__(self):
+        if not self.is_policy_enforcement_enabled():
+            self._log_policy(msg="Policy enforcement is not enabled.")
+            return
+
+        self.policy = self.__get_policy()
+
+    @staticmethod
+    def _log_policy(msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
```

Comment:
I would suggest _log_policy_info or something similar. 
_log_policy() sounds like we're logging the policy attribute

Reply:
how about _log_policy_event()? _log_policy_info might make it seem like we're just writing to logger.info, but we're also writing to logger.error

Reply:
Ah I see, _log_policy_message? event is also ok with me